agents/loco_mc_agent.py

```diff
@@ -284,14 +281,15 @@ def controller_step(self):
                 self.perceive(force=True)
             # change this to memory.get_time() format?
             self.last_chat_time = time.time()
+            # For now just process the first incoming chat, where chat -> [speaker, chat]
+            chat_parse = self.chat_parser.get_parse(incoming_chats[0][1])
```

Comment:
sorry, I don't understand, shouldn't this, and the whole block, be in the .perceive()?